### PR TITLE
Peregrine updates

### PR DESCRIPTION
**Pull request by cbeighley**
- Update Python requirements to specific version numbers
- Add support to cython `NavMsg` class for comparison and pickling
- Fix some `__cinit__` methods

### CODE REVIEW

File: python/swiftnav/nav_msg.pyx [hunk 1]
```diff
@@ -40,3 +43,57 @@ cdef class NavMsg:
   def process_subframe(self, e):
```
Comment (dt-exafore) [inline python/swiftnav/nav_msg.pyx]: Please, take into account that In order to implement L2C capability monitor (please, see PR https://github.com/swift-nav/libswiftnav/pull/294/ still not merged) I updated process_subframe. Now it takes pointer to gps_l1ca_decoded_data_t as the 2nd parameter. So I guess tmp type should be changed. However, I believe this should be done after my PR be merged.

File: python/swiftnav/nav_msg.pyx [hunk 2]
```diff
@@ -40,3 +43,59 @@ cdef class NavMsg:
...
+    http://docs.cython.org/src/userguide/special_methods.html.
+    """
+    if op == 2:
+      return self._equal(other)
+    elif op == 3:
+      return not self._equal(other)
+    else:
+      raise NotImplementedError
+
+  def _equal(self, other):
```
Comment (mookerji) [inline python/swiftnav/nav_msg.pyx]: Needs a test!

File: python/swiftnav/nav_msg.pyx [hunk 3]
```diff
@@ -31,6 +31,9 @@ cdef class NavMsg:
   def from_dict(self, d):
     self._thisptr = d
 
+  def __reduce__(self):
```
Comment (mookerji) [inline python/swiftnav/nav_msg.pyx]: Needs a test!

File: python/swiftnav/track.pyx [hunk 4]
```diff
@@ -236,19 +237,18 @@ cdef class AidedTrackingLoop:
...
-                  self._thisptr.code_bw,
-                  self._thisptr.code_zeta,
-                  self._thisptr.code_k,
-                  self._thisptr.carr_to_code,
-                  self._thisptr.carr_freq,
-                  self._thisptr.carr_bw,
-                  self._thisptr.carr_zeta,
-                  self._thisptr.carr_k,
-                  self._thisptr.carr_freq_b1)
+                  kwargs['loop_freq'],
```
Comment (mookerji) [inline python/swiftnav/track.pyx]: I might do something like `kwargs.pop('loop_freq', <some default or nothing>)` here?

File: python/swiftnav/track.pyx [hunk 5]
```diff
@@ -396,10 +404,10 @@ cdef class CN0Estimator:
 
   def __cinit__(self, **kwargs):
     cn0_est_init(&self._thisptr,
-                 self._thisptr.bw,
-                 self._thisptr.cn0_0,
-                 self._thisptr.cutoff_freq,
-                 self._thisptr.loop_freq)
+                 kwargs['bw'],
```
Comment (mookerji) [inline python/swiftnav/track.pyx]: Same pop pop pop stuff.

File: python/swiftnav/track.pyx [hunk 6]
```diff
@@ -396,10 +404,10 @@ cdef class CN0Estimator:
...
+                 kwargs['cn0_0'],
+                 kwargs['cutoff_freq'],
+                 kwargs['loop_freq'])
 
   def update(self, I, Q):
     """
```
Comment (mookerji) [inline python/swiftnav/track.pyx]: In general, I don't think the tracking stuff in libswiftnav python has tests, so adding some sanity tests would be great.

**Pull request closed without merging**